django/http/multipartparser.py

```diff
@@ -166,15 +173,30 @@ def parse(self):
                 field_name = force_text(field_name, encoding, errors='replace')
 
                 if item_type == FIELD:
+                    # Avoid storing more than DATA_UPLOAD_MAX_NUMBER_FIELDS
+                    num_post_keys += 1
+                    if (settings.DATA_UPLOAD_MAX_NUMBER_FIELDS is not None and
+                            settings.DATA_UPLOAD_MAX_NUMBER_FIELDS < num_post_keys):
+                        raise SuspiciousOperation('Too many fields')
```

Comment:
+1 about improving the messages so a developer can better understand. I thought of this after my review yesterday. Subclasses of `SuspiciousOperation` seem okay to me too.